<launch>

  <!-- Set these parameters in a higher level launch file -->
  <arg name="control_command_topic"/>
  <arg name="map_frame"/>
  <arg name="trajectory_topic"/>
  <arg name="vehicle_frame"/>

  <!-- Set all default values in this yaml file -->
  <arg name="config" default="$(find kal_controller_ros_tool)/launch/params/ControllerInterface.yaml"/>

  <node pkg="kal_controller_ros_tool" type="controller" name="controller" ns="/control" output="screen">
    <rosparam command="load" file="$(arg config)"/>
    <param name="/control_command_publisher_topic" value="$(arg control_command_topic)"/>
    <param name="/map_frame" value="$(arg map_frame)" />
    <param name="/trajectory_subscriber_topic" value="$(arg trajectory_topic)" />
    <param name="/vehicle_frame" value="$(arg vehicle_frame)" />
  </node>

</launch>
